src/VisualStudio/Core/Impl/CodeModel/ProjectCodeModelFactory.cs

```diff
@@ -199,15 +182,14 @@ public EnvDTE.FileCodeModel GetOrCreateFileCodeModel(ProjectId id, string filePa
             => GetProjectCodeModel(id).GetOrCreateFileCodeModel(filePath).Handle;
 
         public void ScheduleDeferredCleanupTask(Action a)
-            => _deferredCleanupTasks.Add(_threadingContext.JoinableTaskFactory.StartOnIdle(a, VsTaskRunContext.UIThreadNormalPriority));
-
-        void IDisposable.Dispose()
         {
-            // Stop any outstanding BG work we've queued up.
-            _cancellationTokenSource.Cancel();
-
-            // Now wait for all our cleanup tasks to finish before we return.
-            _deferredCleanupTasks.Join();
+            var joinableTask = _threadingContext.JoinableTaskFactory.StartOnIdle(a, VsTaskRunContext.UIThreadNormalPriority);
```

Comment:
yikes.  not your issue, but i'm fairly shocked we issue a request to both run on the ui thread and at normal pri.